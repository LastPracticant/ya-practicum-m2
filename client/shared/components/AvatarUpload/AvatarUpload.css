.input-upload {
    display: none;
    cursor: pointer;
}

.avatar-upload {
    background-color: red !important;
}
