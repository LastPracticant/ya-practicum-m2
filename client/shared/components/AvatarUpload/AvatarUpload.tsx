import React from 'react';
import { Avatar, InputProps } from '@material-ui/core';
import './AvatarUpload.css';
import { AVATAR_DEFAULT } from 'client/shared/consts';

export interface InputAvatarProps extends InputProps {
    name: string;
    src?: string;
    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const AvatarUploadComponent = React.forwardRef<
HTMLInputElement,
InputAvatarProps
>(({ name, src, onChange }, ref) => (
    <label htmlFor={`upload_${name}`}>
        <input
            ref={ref}
            id={`upload_${name}`}
            name={name}
            onChange={onChange}
            type="file"
            className="input-upload"
        />
        <Avatar className="avatar-upload" src={src}>
            {AVATAR_DEFAULT}
        </Avatar>
    </label>
));

export const AvatarUpload = React.memo(AvatarUploadComponent);
