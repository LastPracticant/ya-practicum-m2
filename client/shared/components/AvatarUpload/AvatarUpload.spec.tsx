import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { AvatarUpload } from './AvatarUpload';

afterEach(cleanup);

describe('AvatarUpload', () => {
    it('Render AvatarUpload component', () => {
        const { asFragment } = render(<AvatarUpload name="login" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Render AvatarUpload component with image', () => {
        const urlImg = 'https://million-wallpapers.ru/wallpapers/5/57/459836597849189/rembo.jpg';
        const { asFragment } = render(<AvatarUpload src={urlImg} name="login" />);
        expect(asFragment()).toMatchSnapshot();
    });
});
