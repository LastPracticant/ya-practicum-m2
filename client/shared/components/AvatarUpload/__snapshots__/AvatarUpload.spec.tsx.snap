// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarUpload Render AvatarUpload component 1`] = `
<DocumentFragment>
  <label
    class="upload-avatar"
    for="upload_login"
  >
    <input
      id="upload_login"
      name="login"
      type="file"
    />
    <div
      class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
    >
      LP
    </div>
  </label>
</DocumentFragment>
`;

exports[`AvatarUpload Render AvatarUpload component with image 1`] = `
<DocumentFragment>
  <label
    class="upload-avatar"
    for="upload_login"
  >
    <input
      id="upload_login"
      name="login"
      type="file"
    />
    <div
      class="MuiAvatar-root MuiAvatar-circle"
    >
      <img
        class="MuiAvatar-img"
        src="https://million-wallpapers.ru/wallpapers/5/57/459836597849189/rembo.jpg"
      />
    </div>
  </label>
</DocumentFragment>
`;
